xarray/backends/netCDF4_.py

```diff
@@ -41,6 +41,11 @@ def __init__(self, variable_name, datastore):
             dtype = np.dtype('O')
         self.dtype = dtype
 
+    def __setitem__(self, key, value):
+        with self.datastore.ensure_open(autoclose=True):
+            data = self.get_array()
+            data[key] = value
+
```

Comment:
@shoyer, is this what you were describing in https://github.com/pydata/xarray/issues/1464#issuecomment-341329662

Reply:
Yes, this looks right to me.